Identifier.ts

```diff
@@ -25,14 +25,33 @@ export namespace Identifier {
 	export function toUint48(identifier: Identifier): number {
 		return Number.parseInt(toHexadecimal(identifier, 12), 16)
 	}
-	export function fromBinary(identifier: Uint8Array): Identifier {
-		return Base64.encode(identifier, "url")
+	export function fromBinary(identifier: Uint8Array, standard: Base64.Standard = "url"): Identifier {
+		return Base64.encode(identifier, standard)
 	}
 	export function toBinary(identifier: Identifier): Uint8Array {
 		return Base64.decode(identifier, "url")
 	}
-	export function generate(length: Length): Identifier {
-		return fromBinary(crypto.getRandomValues(new Uint8Array((length / 4) * 3)))
+	export function generate(length: Length): Identifier
+	export function generate(
+		length: Length,
+		ordering: Extract<Base64.Standard, "ordered" | "reversed">,
+		value?: number | Uint8Array | string
+	): Identifier
+	export function generate(
+		length: Length,
+		ordering?: Extract<Base64.Standard, "ordered" | "reversed">,
+		value?: number | Uint8Array | string
+	): Identifier {
+		let result: Identifier | undefined = undefined
+		if (value || value == 0 || value == "")
+			result = Base64.encode(value, ordering).substring(0, length)
+		return (
+			(result ?? "") +
+			fromBinary(crypto.getRandomValues(new Uint8Array((length / 4) * 3)), ordering).substring(
+				0,
+				length - (result ? result.length : 0)
+			)
+		)
```

Comment:
This won't care for unused bits. Most values will generate more bits than it needs, can we try to truncate this? Or is this up to the implementation around this function?

Reply:
If you're using the overload you probably want as many bits as needed (and fit in the length) for it.
You will throw away some stuff from the fromBinary() call, but I couldn't come up with a clever way to get the correct lengths another way.

Reply:
It's not obvious how to do it since we don't want to generate ids that wont be in order when more bits are needed. Lets not solve this now and go back and rethink this at a later stage. Adding this functionality should be compatible with the current code. Since before the improvement we have `ABCDZX_1X_2X_3` (`ABCD` represents the prefix, `Z` represents the extra bits of zeros and `X_n` is a randomly generated base64 character.) and after `ABCDX_1X_2X_3X_4` (`Z` has been replaced by a random character and should preserve ordering.).